### PR TITLE
Add explicit links to the GCSE sections

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

This adds in new and create actions/views/routes for all the GCSE controllers. This allows us to pass in explicit link to all the backlinks.

This ones a bit more complex than most of the others on quite a few views the backlink could go back to multiple locations based on whether its an international qual or if they input a gcse with a grade of D or worse.

## Changes proposed in this pull request

- Add a new and create action to each controller
- Add the routes
- Add the new view
- Ensure backlinks have explicit values passed in 

## Guidance to review

Per commit is the way to go. It's 1 commit per controller

I've given it a pretty thorough manual test and it seems to be behaving, but it's worth double checking

## Link to Trello card

https://trello.com/c/GuiJ30Jg/3397-catch-all-give-back-links-explicit-targets

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by davidgisbey** [comment]
Hey Frankie, just pinging you because i know how much you love backlinks haha

**Review by stevehook — COMMENTED** [review]
A few odd things happened on the review app, not sure how much they had to do with these changes (or me deleting rows from the `application_qualifications` table for test purposes):

- I couldn't get the Back button to work differently for new qualifications - it always took me back to the review page. e.g. Add new English GCSE, fill in type form and save, fill in type form and save then hit back - returns me to the review page.
- I couldn't say _No, I’ll come back to it later_ without having a valid qualification. E.g. if year was blank then I get the validation I'd expect when selecting _Yes, I have completed this section_
image

Will carry on review later...

**Comment by davidgisbey** [comment]
Thanks for the review Steve. It looks like the second issue is prexisting, but youre right it's undesirable. Will sort Tuesday 👍 

Have a great long weekend!

**Review by stevehook — COMMENTED** [review]
A few questions/suggestions inline

**Comment by davidgisbey** [comment]
Thanks for all the feedback Steve. 

1. I can't seem to replicate the error
2. This is a pre-existing issue. I'll card it up!

File: app/controllers/candidate_interface/gcse/grade_explanation_controller.rb [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+    def new
+      set_previous_path
+      @form = CandidateInterface::GcseGradeExplanationForm.build_from_qualification(current_qualification)
+    end
+
+    def create
+      @form = CandidateInterface::GcseGradeExplanationForm.new(update_params)
+
+      if @form.save(current_qualification)
+        redirect_to candidate_interface_gcse_details_edit_year_path(params[:subject])
```
Comment (stevehook) [inline app/controllers/candidate_interface/gcse/grade_explanation_controller.rb:12]: Missed this on the first look. I think this messes up the back button on the year form following this form.
```suggestion
        redirect_to candidate_interface_gcse_details_new_year_path(params[:subject])
```

**Review by stevehook — APPROVED** [review]
One more thing I missed first time round. Otherwise worked perfectly when I tested all the permutations I could think of. Great stuff!

**Pull request merged**